cpp/open3d/pipelines/integration/UniformTSDFVolume.cpp

```diff
@@ -138,6 +154,64 @@ std::shared_ptr<geometry::PointCloud> UniformTSDFVolume::ExtractPointCloud() {
     return pointcloud;
 }
 
+std::shared_ptr<geometry::TriangleMesh>
+UniformTSDFVolume::ExtractTriangleMesh_v2() {
+    auto mesh = std::make_shared<geometry::TriangleMesh>();
+    //float *image_array = new float[resolution_  * resolution_ * resolution_];
+    //const long r_sq = resolution_ * resolution_;
+    int dims[] = {resolution_, resolution_, resolution_};
+    float ***values = new float **[dims[0]];
```

Comment:
All pointers are allocated but not released. It's better to allocate a flattened std::vector and unroll the indices.